### ISSUE TITLE
MQ service controller

### ISSUE THREAD

**Issue by tabern** [aws-controllers-k8s/community#390]
## New ACK Service Controller

Support for [Amazon MQ](https://aws.amazon.com/amazon-mq/)

### List of API resources

List the API resources in order of importance to you:

* [Broker](https://docs.aws.amazon.com/amazon-mq/latest/api-reference/brokers.html#CreateBroker)
* [Configuration](https://docs.aws.amazon.com/amazon-mq/latest/api-reference/configurations.html#CreateConfiguration)

**Comment by mathewpeterson** [comment]
Would this be different than the new AWS RabbitMQ offering?

**Comment by jaypipes** [comment]
> Would this be different than the new AWS RabbitMQ offering?

Yes. The RabbitMQ offering is simply a feature of the existing Amazon MQ service. The ACK service controller for Amazon MQ will allow Kubernetes users to create Amazon MQ clusters (of whatever underlying engine, rabbitmq or otherwise) using Kubernetes manifests.

**Comment by jaypipes** [comment]
We're going to work on the Broker resource [for developer preview](https://github.com/aws/aws-controllers-k8s/milestone/12), aiming for RabbitMQ support before tackling the trickier Configuration and User resources of the AmazonMQ API.

See #624

**Comment by shreben** [comment]
Hello team
First, thank you for the efforts put into the ACKs.
I am just wondering is there any estimations on when the Broker Configuration will be added to MQ ACK?
Also maybe you can advise a proper way of handling a configuration of a broker that was created with the ACK. I am currently looking into building a network of brokers and this involves per broker configuration.
Thank you.

**Comment by A-Hilaly** [comment]
Hi @shreben - no timeline regarding new features for `mq-controller` yet. However since ACK controllers a mostly generated you can definitely give it a shot and try to generate Configuration support. Happy to answer with any questions you have. We are available at #aws-controllers-k8s channel in k8s slack.
